weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts

```diff
@@ -501,6 +425,16 @@ const fetchEvaluationComparisonResults = async (
   // 4. Populate the predictions and scores
   const evalTraceRes = await evalTraceResProm;
 
+  const isImperativeEval = evalTraceRes.calls.some(isImperativeEvalCall);
```

Comment:
i think you are making the incorrect assumption that all evaltraceres evals are either imperative or non-imperative. This is not a solid assumption. I think you need to split this into 2 groups for processing separately